Elf/__init__.py

```diff
@@ -907,12 +900,44 @@ def getDynSymTabInfo(self):
         in self.dynsymtabct.  Perhaps this is horrible and should be stricken
         from the code.
         '''
-        #FIXME: make this use both SECTION and PT_DYNAMICS versions...
+        # if DT_SYMTAB doesn't exist, we don't have a symbol table defined by DYNAMICS
         symtabva = self.dyns.get(DT_SYMTAB)
         if symtabva is None:
             return None, None, None
 
         symsz = self.dyns.get(DT_SYMENT)
+
+        # calculate/reuse symbol count
+        if not self.dynsymtabct:    # check if we've already got this value
+            # calculate the Dynamic Symbol Table Size
+            self.dynsymtabct = len(self.dynstrtab)  # cheat: there is a 1-to-1 relationship between symbols and strings in these tables
+            if not self.dynsymtabct:
+                # if we don't 
+                self._parseDynStrs()
+                self.dynsymtabct = len(self.dynstrtab)  # cheat: there is a 1-to-1 relationship between symbols and strings in these tables
```

Comment:
How sure are you of this relationship? I ask because in the function docstring, there is this line: 
```
Supposedly there is a 1:1 relationship between DynSyms and DynStrs, but that can be misleading
```
Which has me concerned about the assumptions here.

Reply:
this is not an assumption we can support any more.  there's supposed to be a 1:1, but it's not always true.  that's why this bugfix exists.

Reply:
Right, I'm just pointing out that your comment here says something that seems to be in contradiction to the function docstring. One of them is wrong, I'd just like  one of them to be updated to reflect the bugfix.

Reply:
got it.  removed :)